### PR TITLE
Add sidebarUserPanel

### PR DESCRIPTION
**Pull request by averissimo**
# Implements shinydashboard function `sidebarUserPanel`

This branch is rebased with `develop` so it can be merged

Relevant issues:

- #182

What was changed:

- [x] Style the margins of the element
- [x] Adds documentation
- [x] Removes `sidebarUserPanel` from `not_implemented` list in tests
- [x] Regenerate documentation (man & NAMESPACE

## Open questions

**Q:** Is there a better way to generate padding values? 

Values `padding: .92857143em 1.14285714em;` are taken from Semantic UI css and they translate to `16px`

**Q:** in `R/semantic_dashboard.R` there is a different implementation of `dashboard_sidebar` between branches

The WIP branch does it via `do.call` that avoided using `..1[-1]`, but the implementation on the `develop` branch is newer. The WIP seems cleaner, but I can change it.

## Example

image

### CODE REVIEW

File: .Rbuildignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
+^renv$
```
Comment (osenan) [inline .Rbuildignore]: why do we need `renv` folder and `renv.lock` file here?

File: inst/semantic.dashboard.css [hunk 2]
```diff
@@ -259,3 +259,7 @@ body.no-margin .ui.top.menu.dashboard-header {
   border-top-width: 1px;
 }
 
+.ui.sidebar .user-panel {
+  min-height: 65px;
+  padding: .92857143em 1.14285714em;
```
Comment (osenan) [inline inst/semantic.dashboard.css]: As you put in the description, using `em` looks like a magic number. I think we can have similar effect by using `padding: 10px 10px;` which looks more easy to understand

**Review by osenan — CHANGES_REQUESTED** [review]
Hi, good job! It looks very good

Let me answer some of your questions: The change in the `dashboard_sidebar` using `do.call` is needed to correctly add more elements as html and not as a list. I agree of keeping this approach. Indeed it will benefit the usability of the sidebar in general.

Regarding the padding of the sidebar user panel, I think that it will be easier to maintain to use `px` instead of `em` as suggested in my comment. If user needs to change font size by css then they can also change the padding of the sidebar user panel if it does not look good.

Please change the padding and answer my comment regarding the `renv` I think we do not need it.

Finally, can you confirm that the image we upload is a free image?

File: styles/layout/_sidebar.scss [hunk 3]
```diff
@@ -24,3 +24,8 @@ $opposite-sides: ("left": "right", "right": "left", "top": "bottom", "bottom": "
     border-#{$opposide-side}-width: 1px;
   }
 }
+
+.ui.sidebar .user-panel{
+  min-height: 65px;
+  padding: .92857143em 1.14285714em;
```
Comment (osenan) [inline styles/layout/_sidebar.scss]: same thing than in the `css` file, better use `px` can be set to 10

File: .Rbuildignore (hunk 1 above)
Comment (averissimo) [reply .Rbuildignore]: You're right, it sneaked in here as I'm using renv to manage the local environment. Sorry! I will remove it

File: inst/semantic.dashboard.css (hunk 2 above)
Comment (averissimo) [reply inst/semantic.dashboard.css]: Done!

File: styles/layout/_sidebar.scss (hunk 3 above)
Comment (averissimo) [reply styles/layout/_sidebar.scss]: Done!

**Comment by averissimo** [comment]
@osenan Sorry for taking this long to reply to the review.

**👉 BLUF: I think I resolved all your pending comments, let me know if I can merge the PR, or if you request any further changes**

I think I resolved all your pending comments:

* The `renv` sneaks in a lot when using it locally to manage versions.
* Change the em to px
    * Updated SASS call to reflect current library calls

Regarding the image, it has a Free to Use license 

Photo from: https://www.pexels.com/photo/silhouette-of-person-standing-on-beach-during-sunset-3492736/

https://www.pexels.com/license/

image

**Review by osenan — APPROVED** [review]
Great! And thank you for explanation

**Pull request merged**